classes/install/Upgrade.inc.php

```diff
@@ -1235,33 +1235,83 @@ function migrateDefaultUsageStatistics() {
 
 		return true;
 	}
-	
+
 	/**
-	 * For 2.4.6 upgrade: to enable localization of a CustomBlock, 
+	 * For 2.4.6 upgrade: to enable localization of a CustomBlock,
 	 * the blockContent values are converted from string to array (key: primary_language)
 	 */
 	function localizeCustomBlockSettings() {
 		$pluginSettingsDao = DAORegistry::getDAO('PluginSettingsDAO');
 		$journalDao = DAORegistry::getDAO('JournalDAO');
 		$journals = $journalDao->getJournals();
-		
+
 		while ($journal = $journals->next()) {
 			$journalId = $journal->getId();
 			$primaryLocale = $journal->getPrimaryLocale();
-			
+
 			$blocks = $pluginSettingsDao->getSetting($journalId, 'customblockmanagerplugin', 'blocks');
-			foreach($blocks as $block) {
+			if ($blocks) foreach ($blocks as $block) {
 				$blockContent = $pluginSettingsDao->getSetting($journalId, $block, 'blockContent');
-				
+
 				if (!is_array($blockContent)) {
 					$pluginSettingsDao->updateSetting($journalId, $block, 'blockContent', array($primaryLocale => $blockContent));
 				}
 			}
 			unset($journal);
 		}
-		
+
 		return true;
 	}
+
+	/**
+	 * For 2.4.6 upgrade: Remove the "Custom Identifier" suffix option in the DOI and URN plugin and
+	 * use the ctustom suffix pattern %x instead.
+	 * @return boolean
+	 */
+	function removeCustomIdentifierSuffixOption() {
+		$pluginSettingsDAO =& DAORegistry::getDAO('PluginSettingsDAO');
+		$journalDao =& DAORegistry::getDAO('JournalDAO');
+		$journals =& $journalDao->getJournals();
+		while ($journal =& $journals->next()) {
```

Comment:
Don't forget the unset($journal) at the end of the loop -- forgetting those can sometimes lead to hard-to-debug problems (though I think in this case it would be OK)